force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -126,16 +154,21 @@ public with sharing class ClassesObjectsMethods {
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
         // The variable to store the concatenated string
-
+        String scientistNamesWithG = '';
         // Loop through the list of scientists' names
-
+        for (String name : scientistNames){
             // If the name doesn't contain the letter 'G', skip this iteration
-
+            if (!name.toLowerCase().contains('g')){
+                continue;
+            }
             // Add the name to the result string, followed by a comma
-
+            scientistNamesWithG += name + ',';
+        }
         // Remove the last comma from the result string
-
-        return null; // Replace null with the variable you used to store the result
+        if(scientistNamesWithG.endsWith(',')){
+            scientistNamesWithG = scientistNamesWithG.substring(0, scientistNamesWithG.length() - 1);
+        }
+        return scientistNamesWithG; 
```

Comment:
No need to use the continues statement here. If your loop variable does not meet the condition of containing 'g' you will continue to the next element in the collection regardless. 

Take a look at the String.removeEnd method.